### PR TITLE
Add "Clear Cache" Button 

### PR DESCRIPTION
**Pull request by 00travelgirl00**
PR includes just a mockup, not a  functionality!

### CODE REVIEW

**Comment by 00travelgirl00** [comment]
Screenshot of the changes 

bildschirmfoto 2018-02-05 um 22 58 52

File: inc/cachify.settings.php [hunk 1]
```diff
@@ -112,3 +112,19 @@
...
+<form method="post" action="options.php">
+	<?php settings_fields( 'cachify' ) ?>
+    <table class="form-table">
+        <tr>
+            <th scope="row">
+				<?php esc_html_e( 'Delete Caching', 'cachify' ) ?>
+            </th>
+            <td>
+                <label for="delete-cache">
+                    <button type="button" class="button-secondary">Clear Cache</button>
```
Comment (websupporter) [inline inc/cachify.settings.php]: esc_html_e( 'Clear Cache', 'cachify' );

**Comment by websupporter** [comment]
Thanks @00travelgirl00,
so, we would basically need to add the actual "cache-flush"-functionality, right?

If you don't feel comfortable to add it, I can do that. Other than that, why is this "option" not together with the other options? Maybe because its not an option, but to me it seems a bit lost down there 🤔

**Comment by krafit** [comment]
I'm not entirely sure, if this is fixing an actual problem 🤔 
I don't see an immediate benefit of adding another option and would rather add a brief description text alongside the *Cache expiration* option.

**Comment by 00travelgirl00** [comment]
@krafit I understand the "Cache expiration" as a automatic time period when the cache should be renewed. 

The Use case for the "Clear Cache" Button is, if I made a change but can't see it on the website. The first thing I want to do is to make sure its not the cache, so I would like to clear it to check, if the setting/code what ever is wrong or not. 

@websupporter I put it below, because its not a actual setting, that I could save. It is an other functionality, so I thought its better, to put it on an extra space. 

I made different mockups, also the "Clear Cache" Button next to the Save Changes Button, but I tought it could might be confusing for non experienced users. 

But we could place it above the links.

**Comment by websupporter** [comment]
@krafit 
right now, we only have the trash icon on the top right. This might not be super obvious for new users. My first guess would also be to go to some settings page to flush the cache, which is something else than the cache expiration.

@00travelgirl00 
maybe we could just add a "flush cache now"-button underneath the hours-input field of the cache expiration? Around here: https://github.com/00travelgirl00/cachify/blob/e9cb0376b96497494326c7eab2e9c156716d73ae/inc/cachify.settings.php#L37-L38

**Comment by stklcode** [comment]
> right now, we only have the trash icon on the top right

Wow, using the plugin for years, knowing the code and still never noticed that button as part of Cachify (or maybe forgot about it). So much for "not super obvous"... :open_mouth: Maybe move it to the dashboard widget next to the cache size thinking

I'd personally like to go even further and rework the call to `flush_total_cache(true)` call in die `validate_options()` method in favor of the proposed new button. (not point of this PR, but should be taken into count)
It's rather annoying that increasing the caching timeout or changing e.g. comment settings flushes all pre-existing values...

Flushing on backend change, drecrease of timeout and update of the blacklist is obviously still a good idea. And of course, making no-change save to flush works, but is not intuitive

If you know it's flushing on save OK, but on the UX perspective not perfect though. And yes, I'm with @00travelgirl00 to separate the button into the settings form, as it is basically no setting.

**Comment by chesio** [comment]
Related: #102

I think the button (no matter where it ends up) would benefit from AJAX-ification.

In https://github.com/chesio/cachify/tree/issue-102, I have an AJAX solution for the not-super-obvious button in admin bar :-)

**Comment by 2ndkauboy** [comment]
@websupporter @krafit looks good to me, ready to merge.

**Pull request merged**